force-di/main/classes/di_Binding.cls

```diff
@@ -182,11 +183,28 @@ public abstract class di_Binding implements Comparable {
             }
             // Filter bindings returned by preconfigured criteria
             List<di_Binding> matchedBindings = new List<di_Binding>();
-            for(di_Binding bind : bindings) {
-                if(developerName!=null && bind.DeveloperName!=null && bind.DeveloperName.equalsIgnoreCase(developerName)) {
-                    matchedBindings.add(bind);
-                } else if (bindingObject!=null && bind.BindingObject == bindingObject) {
-                    matchedBindings.add(bind);
+            for (di_Binding bind : bindings) {
+                // if both filtering parameters were specified
+                if ( String.isNotBlank(this.developerName) && this.bindingObject != null ) 
+                {
+                    // then require both match for the binding to be included.
+                    if ( this.developerName.equalsIgnoreCase( bind.DeveloperName )
+                        && this.bindingObject == bind.BindingObject )
+                    {
+                        matchedBindings.add(bind);
+                    }
+                }
+                else {
+                    // else match on any of the available filtering parameters
+                    if (String.isNotBlank(this.developerName) && String.isNotBlank(bind.DeveloperName)
+                            && bind.DeveloperName.equalsIgnoreCase(developerName)) 
+                    {
+                        matchedBindings.add(bind);
+                    } 
+                    else if (bindingObject != null && bind.BindingObject == bindingObject) 
```

Comment:
@ImJohnMDaniel Line 204, for consistency I'd prefix `bindingObject` as `this.bindingObject` as done earlier in the method

Reply:
Corrected